src/lib/stateBasedTestHelper.ts

```diff
@@ -185,7 +173,7 @@ export async function testSagaWithState<StateT, Payload>(
       awaitingMessages = awaitingMessages.filter((config) => !isType(action, config.actionCreator));
     },
     spawn: <P extends any[], T>(
-      funcOrBoundEffect: BoundEffect<Environment<StateT, Action>, P, T> | SagaFunc<StateT, Action, P, T>,
+      funcOrBoundEffect: BoundEffect<SagaEnvironment<StateT>, P, T> | FuncWithEnv<StateT, P, T>,
```

Comment:
The types on these are just a leftover, we don't need them anymore, right?

Reply:
Yeah, I will remove them tomorrow.

Reply:
Removed these already. In the app we have an issue with the overload for `run` (`spawn` would be the same) in that I can't call `run` without providing the `args` (the never is not correctly inferred.


You can see the issue when adding these lines to the end of `types.ts`:

```ts
const sagaEnv: SagaEnvironment<any> = null as any;
const boundEffect: BoundEffect<SagaEnvironment<any>, [number, number], boolean> = null as any;
sagaEnv.spawn(boundEffect);
```

It should work without the params provided another time, but TS requires one to provide 2 numbers here.

Reply:
Okay, I will have a look. But if you pass in a bound effect, the `run`/`spawn` shouldn't receive any additional arguments anymore, correct?

Reply:
Yes, not having to pass the arguments again would be the desired state. I achieved this for `run` as a workaround by adding an overload (see other comment on commit) just to get this working in OMS for now.

But the overload then required to add explicit types on the implementations again, so it's not ideal. I was searching whether there is an open TypeScript issue for this, but was unable to find any. So far I see no reason why it shouldn't work, but somehow the types get lost / not evaluated correctly.